include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -2343,6 +2343,42 @@ __pattern_sort(_ExecutionPolicy&& __exec, _RandomAccessIterator __first, _Random
     });
 }
 
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _RandomAccessIterator2,
+          typename _Compare, typename _IsVector>
+oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy, void>
+__pattern_sort_by_key(_ExecutionPolicy&& __exec, _RandomAccessIterator1 __keys_first,
+                      _RandomAccessIterator1 __keys_last, _RandomAccessIterator2 __values_first, _Compare __comp,
+                      _IsVector /*vector=*/, /*is_parallel=*/::std::false_type) noexcept
+{
+    auto __beg = oneapi::dpl::make_zip_iterator(__keys_first, __values_first);
+    auto __end = __beg + (__keys_last - __keys_first);
+    auto __cmp_f = [__comp](const auto& __a, const auto& __b) { return __comp(::std::get<0>(__a), ::std::get<0>(__b));};
+
+    ::std::sort(__beg, __end, __cmp_f);
+}
+
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _RandomAccessIterator2,
+          typename _Compare, typename _IsVector>
+oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy, void>
+__pattern_sort_by_key(_ExecutionPolicy&& __exec, _RandomAccessIterator1 __keys_first,
+                      _RandomAccessIterator1 __keys_last, _RandomAccessIterator2 __values_first, _Compare __comp,
+                      _IsVector /*vector=*/, /*is_parallel=*/::std::true_type)
+{
+    static_assert(::std::is_move_constructible_v<_RandomAccessIterator1> && 
+        ::std::is_move_constructible_v<_RandomAccessIterator2>,
+        "The keys abd values should be move constructible in case of parallel execution.");
```

Comment:
The static assert here is for the iterator and not for the key/value type. Is that intended? Also "keys abd value" should be "keys and value"

Reply:
@adamfidel , could you please re-phrase your question?

Reply:
yes, of course should be a kind of `::std::iterator_traits<_RandomAccessIterator>::value_type`